src/library/scala/util/control/Exception.scala

```diff
@@ -13,24 +13,125 @@ package control
 import scala.reflect.{ ClassTag, classTag }
 import scala.language.implicitConversions
 
-
 /** Classes representing the components of exception handling.
- *  Each class is independently composable.  Some example usages:
+ *
+ *  Each class is independently composable.
+ *
+ *  This class differs from [[scala.util.Try]] in that it focuses on composing exception handlers rather than
+ *  composing behavior.  All behavior should be composed first and fed to a [[Catch]] object using one of the
+ *  `opt`, `either` or `withTry` methods. Taken together the classes provide a DSL for composing catch and finally
+ *  behaviours.
+ *
+ *  === Examples ===
+ *
+ *  Create a `Catch` which handles specified exceptions.
  *  {{{
  *  import scala.util.control.Exception._
  *  import java.net._
  *
  *  val s = "http://www.scala-lang.org/"
+ *
+ *  // Option[URL]
```

Comment:
I think it's more natural to show types inline. `val x: Option[URL] = RHS`. or show a REPL transcript if desirable to show values. I don't know that there is a standard for docs yet, let alone executable examples, but I like the idea of REPL transcripts. I've seen them somewhere. I just saw example at JavaConverters will shows explicit types.